File: game-utils/game-example/bots/java/src/core/api/Response.java
package core.api;

import core.api.commands.Command;
import core.api.commands.Direction;
import core.api.commands.MoveCommand;

import java.util.ArrayList;

public class Response {
    public int __uid;
    private ArrayList<Command> commands = new ArrayList<>();

    public void moveUnit(Direction direction) {
        commands.add(new MoveCommand(direction));
    }
}

File: game-utils/game-example/bots/kotlin/src/core/api/Response.kt
package core.api


interface Command

class Response {
    var __uid: Int = 0
    private val commands = ArrayList<Command>()

    fun moveUnit(direction: Direction) {
        commands.add(MoveCommand(direction))
    }
}

enum class Direction {
    LEFT, RIGHT, UP, DOWN
}

class MoveCommand(var direction: Direction) : Command {
    var __type = this.javaClass.simpleName
}